src/absil/ilread.fs

```diff
@@ -2230,7 +2230,15 @@ and byteAsCallConv b =
         elif ccMaxked = e_IMAGE_CEE_CS_CALLCONV_VARARG then ILArgConvention.VarArg 
         else  ILArgConvention.Default
     let generic = (b &&& e_IMAGE_CEE_CS_CALLCONV_GENERIC) <> 0x0uy
-    generic, Callconv (byteAsHasThis b, cc) 
+    let callConv =
+        if cc = ILArgConvention.Default then
+            let hasthis_masked = b &&& 0x60uy
```

Comment:
I understand why this bit of code was written, to basically mimic `byteAsHasThis` behavior, but instead to return the full `ILCallingConv` with its `ILArgConvention` as Default.

However, feels a bit weird to have duplicate logic. Perhaps there is a better way to handle this.

Looking at `ILArgConvention` and `ILThisConvention`, I feel those could easily be int32 enums. Then `ILCallingConv` could be a struct. With those changes, `ILCallingConv` is now 8 bytes, which actually quite perfect. Any reason why they can't be that way?

Reply:
@TIHan Yes, that sounds like a good idea. It'd just be a more invasive change to be done. :)

Reply:
Actually, they may not have to be denims. You can just make them a struct as well.